_enumerable.rb

```diff
@@ -0,0 +1,84 @@
+module Enumerable
+  def my_each
+    each do |ele|
+      yield ele
+    end
+    self
+  end
+
+  def my_each_with_index
+    length.times do |indx|
+      yield self[indx], indx
+    end
+    self
+  end
```

Comment:
Is expected that:
#my_each_with_index returns an enumerator if no block is given

#my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self

#my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self

#my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self